Source/UI/MLCardFormField/Mask/MLCardFormCustomMask.swift

```diff
@@ -6,118 +6,78 @@ import Foundation
-                bufferText?.removeLast()
-            }
-            
-            return finalText ?? String()
-        }
-        
-        finalText?.appendCharWithMask(Character(string), mask: self.mask)
-        
-        bufferText?.append(string)
-        
-        return finalText ?? String()
+    
+    func clear() {
+        value.removeAll()
     }
-}
-
-internal extension String {
     
-    mutating func applyMask(_ mask: String?) {
-        
-        guard let mask = mask, !self.isEmpty else { return }
+    @discardableResult
+    func applyMask(to string: String) -> String {
+        guard !mask.isEmpty, !string.isEmpty else {
+            self.value = string
+            return string
+        }
         
-        var result = String()
-        var c: Character
-        var i = 0
+        let string = removeMask(to: string)
+        let count = string.count
+        var maskedString = ""
+        var character: Character
+        var index = 0
         
         for m in mask {
-            
             if m != "$" && m != "*" {
-                result.append(m)
+                maskedString.append(m)
                 continue
             }
             
             repeat {
-                c = self[self.index(self.startIndex, offsetBy: i)]
-                i = i + 1
-            } while !(m == "$" && c.isNumber) && !(m == "*" && c.isLetter) && i < self.count
-        
-            result.append(c)
+                character = string[string.index(string.startIndex, offsetBy: index)]
+                index += 1
+            } while !(m == "$" && character.isNumber) && !(m == "*" && character.isLetter) && index < count
             
-            if i >= self.count {
+            maskedString.append(character)
+            
+            if index >= count {
                 break
             }
         }
-
-        self = result
-    }
-    
-    mutating func appendCharWithMask(_ character: Character?, mask: String?) {
-        
-        guard let character = character, let mask = mask, self.count < mask.count else { return }
-
-        var i = self.count
-        var c: Character?
-        var result = String()
         
-        repeat {
-            
-            if let ch = c {
-                result.append(ch)
-            }
-            
-            c = mask[mask.index(mask.startIndex, offsetBy: i)]
-            
-            i = i + 1
-            
-        } while c != "$" && c != "*" && i < mask.count
+        self.value = maskedString
+        return maskedString
+    }
     
-        if (c == "$" && character.isNumber) || (c == "*" && character.isLetter) {
-            result.append(character)
-            self.append(result)
-        }
+    @discardableResult
+    func shouldChangeCharactersIn(
+        _ range: NSRange,
+        with string: String
+    ) -> String {
+        let newString = (self.value as NSString).replacingCharacters(in: range, with: string)
```

Comment:
What if the user paste a range of characters on the card number field, like "1234 $$ MEU NOME 2345"?

Reply:
the mask will take away all the characters not allowed. for example `1234 $$ MEU NOME 2345` using the card mask translate into `1234 2345`

Reply:
also, we should add a UT for this

Reply:
Ok, but what if the original mask itself expects a character `\*`? Example, 25.833.555-X2? It will remove the characters?